freeglut/freeglut/src/fg_menu.c

```diff
@@ -780,14 +780,29 @@ void fghCalculateMenuBoxSize( void )
 /*
  * Creates a new menu object, adding it to the freeglut structure
  */
-int FGAPIENTRY glutCreateMenu( FGCBMenu callback )
+int FGAPIENTRY glutCreateMenuUcall( FGCBMenuUC callback, FGCBUserData userData )
 {
     /* The menu object creation code resides in fg_structure.c */
-    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenu" );
+    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenuUcall" );
     if (fgState.ActiveMenus)
         fgError("Menu manipulation not allowed while menus in use.");
 
-    return fgCreateMenu( callback )->ID;
+    return fgCreateMenu( callback, userData )->ID;
+}
+
+/* Standard glutCreateMenu */
+static void glutCreateMenuCallback( int menu, FGCBUserData userData )
```

Comment:
Correct, that is the format I use for all callbacks so that old and new functions continue to work.

* Ah, I see what you're getting it. Yea, I can change the `glutXXXCallback` functions to be `fghXXXCallback`. Didn't think about that (and didn't know the difference between `fg` and `fgh` as I didnt' see it mentioned anywhere).
* I did generalize this for fg_callbacks, but it's a bit ugly and for the functions that don't use it, there are some spots where it can't be generalized. I was able to do it for window callbacks because all of them (except for keyboard callbacks) only use ints. I'll look into it and see what I can do.